### PR TITLE
TieBreaker Modeling Formulation

### PR DESCRIPTION
**Pull request by psinha25**

### CODE REVIEW

File: src/aggregate_poisson.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+DATA_STORE = DATA_STORE = '/home/ps35324/gpu-sharing-scheduler/src/data-store/scheduler.db'
```
Comment (aravindhank11) [inline src/aggregate_poisson.py]: This is an issue. Please avoid using hard coded paths. Can this be given as an input parameter?

File: src/batched_inference.py [hunk 2]
```diff
@@ -156,16 +157,143 @@ def infer(self):
         )
         generation_output = self._model.generate(
             input_ids=input_tokens,
-            max_new_tokens=self._batch_size
+            max_new_tokens=self._batch_size,
+            do_sample=True
```
Comment (aravindhank11) [inline src/batched_inference.py]: Why is the base class modified?  Also seems like this parameter is not being used.

File: run_job_mix.sh [hunk 3]
```diff
@@ -413,8 +420,9 @@ setup_expr()
     ipcrm --all=msg || :
 }
 
+cd ~/llama2-expr
```
Comment (aravindhank11) [inline run_job_mix.sh:423]: Please remove this and make it similar to:
https://github.com/aravindhank11/llama2-expr/blob/ak-live-migration/run_job_mix.sh#L412-L414

Git clone need not be in the home directory always.

File: run_custom_mig_poisson.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
```
Comment (aravindhank11) [inline run_custom_mig_poisson.sh]: Can this file be removed?

File: model-formulation/job_mix_creator.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+DATA_STORE = DATA_STORE = '/home/ps35324/gpu-sharing-scheduler/src/data-store/scheduler.db'
```
Comment (aravindhank11) [inline model-formulation/job_mix_creator.py]: Hard coded Path. Can this be an argument?

File: model-formulation/generate_dataset.py [hunk 6]
```diff
@@ -0,0 +1,1518 @@
+import pandas as pd
+import statistics
+import os
+import shutil
+import ast
+
+# RESULTS_DIR = '../results/a100'
+# RESULTS_DIR = '/home/ps35324/a100'
+RESULTS_DIR = '/home/ps35324/collected_data/'
```
Comment (aravindhank11) [inline model-formulation/generate_dataset.py]: Please don't use hard coded paths.

File: model-formulation/generate_dataset.py [hunk 7]
```diff
@@ -0,0 +1,1518 @@
...
+    count = 0
+    data_points = []
+    for outer_dir in os.listdir(RESULTS_DIR):
+        if os.path.isdir(os.path.join(RESULTS_DIR, outer_dir)):
+            models = outer_dir.split('-')
+            if len(models) == job_size:
+                for sub_dir in os.listdir(os.path.join(RESULTS_DIR, outer_dir)):
+                    if 'closed' in sub_dir:
+                        if len([entry for entry in os.listdir(os.path.join(RESULTS_DIR, outer_dir, sub_dir)) if os.path.isfile(os.path.join(RESULTS_DIR, outer_dir, sub_dir, entry))]) == 16:
+
```
Comment (aravindhank11) [inline model-formulation/generate_dataset.py]: A lot of dead code. Is it even being used?

File: model-formulation/dataset_generator.py [hunk 8]
```diff
@@ -0,0 +1,265 @@
+import pandas as pd
+import sqlite3
+import statistics
+
+
+DATA_STORE = '/home/ps35324/gpu-sharing-scheduler/src/data-store/scheduler.db'
```
Comment (aravindhank11) [inline model-formulation/dataset_generator.py]: Please don't use hard coded paths. Can you please convert them to arguments with defaults and examples?

File: run.sh [hunk 9]
```diff
@@ -30,6 +32,7 @@ run_cmd() {
         sleep 0.01
         ctr=$((ctr+1))
     done
+    sleep 1
```
Comment (aravindhank11) [inline run.sh:35]: All these changes are already in master: https://github.com/aravindhank11/llama2-expr/blob/master/run.sh
Can you please rebase. Something seems wrong

File: run_job_mix.sh [hunk 10]
```diff
@@ -138,19 +138,18 @@ run_orion_expr() {
...
-    model_defn_string="${model_run_params[*]}"
-    ${DOCKER} exec -it ${ORION_CTR} bash -c \
+    duration=120
+    ${DOCKER} exec -it ${orion_ctr} bash -c \
         "LD_PRELOAD='/root/orion/src/cuda_capture/libinttemp.so' \
         python3.8 src/orion_scheduler.py \
         --device-type ${device_type} \
         --duration ${duration} \
-        ${model_defn_string}"
+        --model-details '${model_run_params_raw}'"
```
Comment (aravindhank11) [inline run_job_mix.sh]: All these changes are already in master. Something seems odd. 
https://github.com/aravindhank11/llama2-expr/blob/master/run_job_mix.sh

File: src/batched_inference.py (hunk 2 above)
Comment (psinha25) [reply src/batched_inference.py]: `max_new_tokens` was a parameter already in `batched_inference.py` from your implementation. You were missing a comma after `max_new_tokens=self._batch_size`, and hence I added it.

File: run_custom_mig_poisson.sh (hunk 4 above)
Comment (psinha25) [reply run_custom_mig_poisson.sh]: Ah yes, removing

File: model-formulation/job_mix_creator.py (hunk 5 above)
Comment (psinha25) [reply model-formulation/job_mix_creator.py]: Hmmm, this database contains the data I collected in the previous TieBreaker iteration (over 500K datapoints). Part of it includes the profiles of running each job on each slice in MIG. The file is too large to include in GitHub. For the final artifact, we should have this data formatted and accessible in a better format. For the time being, I'll put a relative path or comment it out completely, as we don't need to run this code any longer to make progress to the deadline.

**Review by aravindhank11 — APPROVED** [review]

**Pull request merged**